src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/ExtraConfiguration/pci/index.js

```diff
@@ -0,0 +1,250 @@
+
+  const { setModifiedFields, setFieldPath, initModifiedFields } =
+    useGeneralApi()
+
+  const { data: hosts = [] } = HostAPI.useGetHostsQuery()
+  const hostPciDevices = hosts.map(getPciDevices).flat()
+
+  const {
+    fields: pciDevices,
+    remove,
+    update,
+    append,
+  } = useFieldArray({
+    name: `${EXTRA_ID}.${TAB_ID}`,
+  })
+
+  useEffect(() => {
+    // Init pci modified fields
+    setFieldPath(`extra.PciDevices.PCI`)
+    initModifiedFields([
+      ...pciDevices.map((element, index) => ({ __aliasPci__: index })),
+    ])
+  }, [])
+
+  /**
+   * Returns an object with the fields that have to be updated in the template.
+   *
+   * @param {object} pci - PCI device data
+   * @returns {object} - Object with the fields that have to be updated
+   */
+  const modifiyFields = (pci) => {
+    if (pci.SPECIFIC_DEVICE) {
+      return setModifiedFields({
+        DEVICE: {
+          __delete__: true,
+        },
+        VENDOR: {
+          __delete__: true,
+        },
+        CLASS: {
+          __delete__: true,
+        },
+        PROFILE: {
+          __delete__: true,
+        },
+        SHORT_ADDRESS: true,
+      })
+    } else {
+      return setModifiedFields({
+        DEVICE: true,
+        VENDOR: true,
+        CLASS: true,
+        PROFILE: true,
+        SHORT_ADDRESS: {
+          __delete__: true,
+        },
+      })
+    }
+  }
+
+  const handleCreatePci = (pci) => {
+    // Set modified fiels to be updated
+    setModifiedFields(modifiyFields(pci))
+    delete pci.SPECIFIC_DEVICE
+
+    // Add to the list
+    append(mapPCINameFunction(pci, pciDevices.length))
+  }
+
+  const handleUpdatePci = (index) => (pci) => {
+    // Set modified fiels to be updated
+    setModifiedFields(modifiyFields(pci))
+    delete pci.SPECIFIC_DEVICE
+
+    // Update the list
+    update(index, mapPCINameFunction(pci, index))
+  }
+
+  const handleRemovePci = (index) => () => {
+    // Set modified fiels to remove the element
+    setFieldPath(`extra.PciDevices.PCI.${index}`)
+    setModifiedFields({ __flag__: 'DELETE' })
+
+    // Remov from the list
+    remove(index)
+  }
+
+  return hostPciDevices ? (
+    <div>
+      <Alert severity="info" variant="outlined" className={classes.warningInfo}>
+        {Tr(T.NicPciWarning)}
+      </Alert>
+      <AttachPciAction
+        oneConfig={oneConfig}
+        adminGroup={adminGroup}
+        onSubmit={handleCreatePci}
+        indexPci={pciDevices.length}
+      />
+      <Stack
+        pb="1em"
+        display="grid"
+        gap="1em"
+        mt="1em"
+        sx={{
+          gridTemplateColumns: {
+            sm: '1fr',
+            md: 'repeat(auto-fit, minmax(500px, 0.5fr))',
+          },
+        }}
+      >
+        {pciDevices?.map((pci, index) => {
+          if (pci?.TYPE === 'NIC') return <></>
+
+          // Search for pci devices in the host
+          const pciHostDevice = find(
+            hostPciDevices,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).